manifests/repo.pp

```diff
@@ -30,63 +30,110 @@
   $username             = undef,
   $password             = undef,
   $description          = undef,
-  $environment          = $::environment,
+  $environment          = "${::environment}", # lint:ignore:only_variable_string
```

Comment:
This stringification is needed for spec tests on Puppet 3.8 & 4.3. In these cases the spec tests doesn't deliver a real string. Error message is:

```Evaluation Error: Error while evaluating a Function Call, :rp_env is not a string.  It looks to be a Symbol```

See the full results here:
https://travis-ci.org/ghoneycutt/puppet-module-yum/builds/168043095